audio-echo/app/src/main/java/com/google/sample/echo/MainActivity.java

```diff
@@ -38,6 +46,8 @@ protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         setContentView(R.layout.activity_main);
 
+        //get run-time permission...
+        requestResourcePermission();
         status_view = (TextView)findViewById(R.id.statusView);
         queryNativeAudioParameters();
```

Comment:
Shouldn't you only do that if the permission is granted?

With the current code path I understand `Boolean permissionGranted` is only set to false after onRequestPermissionsResult, but since this is async, I don't think you have any guarantee that the code will block until the permission is granted.